server/src/test/java/org/opensearch/index/compositeindex/datacube/startree/node/FixedLengthStarTreeNodeTests.java

```diff
@@ -0,0 +1,207 @@
+import org.opensearch.index.codec.composite.datacube.startree.fileformats.writer.StarTreeWriter;
+import org.opensearch.test.OpenSearchTestCase;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.util.ArrayDeque;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Queue;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class FixedLengthStarTreeNodeTests extends OpenSearchTestCase {
+
+    private IndexOutput dataOut;
+    private IndexInput dataIn;
+    private Directory directory;
+
+    @Before
+    public void setup() throws IOException {
+        directory = newFSDirectory(createTempDir());
+    }
+
+    public void test_StarTreeNode() throws IOException {
+
+        dataOut = directory.createOutput("star-tree-data", IOContext.DEFAULT);
+        Map<Long, InMemoryTreeNode> levelOrderStarTreeNodeMap = new LinkedHashMap<>();
+        InMemoryTreeNode root = generateSampleTree(levelOrderStarTreeNodeMap);
+        long starTreeDataLength = StarTreeWriter.writeStarTree(dataOut, root, 7, "star-tree");
+
+        // asserting on the actual length of the star tree data file
+        assertEquals(starTreeDataLength, 247);
+        dataOut.close();
+
+        dataIn = directory.openInput("star-tree-data", IOContext.READONCE);
+
+        StarTreeMetadata starTreeMetadata = mock(StarTreeMetadata.class);
+        when(starTreeMetadata.getDataLength()).thenReturn(starTreeDataLength);
+        when(starTreeMetadata.getDataStartFilePointer()).thenReturn(0L);
+        StarTree starTree = new StarTree(dataIn, starTreeMetadata);
+
+        StarTreeNode starTreeNode = starTree.getRoot();
+        Queue<StarTreeNode> queue = new ArrayDeque<>();
+        queue.add(starTreeNode);
+
+        while ((starTreeNode = queue.poll()) != null) {
+
+            // verify the star node
+            assertStarTreeNode(starTreeNode, levelOrderStarTreeNodeMap.get(starTreeNode.getDimensionValue()));
+
+            Iterator<? extends StarTreeNode> childrenIterator = starTreeNode.getChildrenIterator();
+
+            if (starTreeNode.getChildDimensionId() != -1) {
+                while (childrenIterator.hasNext()) {
+                    StarTreeNode child = childrenIterator.next();
+                    assertStarTreeNode(
+                        starTreeNode.getChildForDimensionValue(child.getDimensionValue(), false),
+                        levelOrderStarTreeNodeMap.get(child.getDimensionValue())
+                    );
+                    queue.add(child);
+                }
+            }
+        }
+
+        dataIn.close();
+
+    }
+
+    private void assertStarTreeNode(StarTreeNode starTreeNode, InMemoryTreeNode treeNode) throws IOException {
+        assertEquals(starTreeNode.getDimensionId(), treeNode.dimensionId);
+        assertEquals(starTreeNode.getDimensionValue(), treeNode.dimensionValue);
```

Comment:
This needs to be expanded to many tests including null, -1, max, min of long values etc.

Lets have some standard tests and even in standard tests lets include variations and in randomized tests skew more towards these cases.